includes/cli/class-co-authors-plus.php

```diff
@@ -15,11 +15,64 @@
  * Co-Authors Plus CLI commands.
  */
 class Co_Authors_Plus {
+	const NEWSPACK_SCHEDULE_AUTHOR_TERM_BACKFILL = 'newspack_cap_author_term_backfill';
+
 	private static $live = false; // phpcs:ignore Squiz.Commenting.VariableComment.Missing
 	private static $verbose = true; // phpcs:ignore Squiz.Commenting.VariableComment.Missing
 	private static $user_logins = false; // phpcs:ignore Squiz.Commenting.VariableComment.Missing
 	private static $guest_author_ids = false; // phpcs:ignore Squiz.Commenting.VariableComment.Missing
 
+	/**
+	 * Add hooks.
+	 */
+	public static function init() {
+		add_action( self::NEWSPACK_SCHEDULE_AUTHOR_TERM_BACKFILL, [ __CLASS__, 'run_cap_cli_command' ] );
+		add_action( 'init', [ __CLASS__, 'register_deactivation_hook' ] );
+	}
+
+	/**
+	 * Execute the co-authors-plus create-terms-for-posts CLI command.
+	 */
+	public static function run_cap_cli_command() {
+		if ( method_exists( 'WP_CLI', 'runcommand' ) ) {
+			$result = WP_CLI::runcommand(
+				'co-authors-plus create-author-terms-for-posts --batched --records-per-batch=50',
+				[
+					'launch' => false, // This keeps any formatting that's been set.
+					'return' => 'all', // This ensures we get stdout, stderr, and return code.
+				]
+			);
+
+			WP_CLI::out( $result->stdout );
+
+			do_action(
+				'newspack_log',
+				$result->return_code,
+				self::NEWSPACK_SCHEDULE_AUTHOR_TERM_BACKFILL,
+				[
+					'type'      => 0 === $result->return_code ? 'success' : 'error',
+					'stdout'    => $result->stdout,
+					'timestamp' => gmdate( 'c', time() ),
+					'file'      => 'newspack_cap_author_terms_backfill',
```

Comment:
According to our `newspack_log` handler, the first arg is an error or debug code string (to let us filter by code in our logging system) and the second arg is a message string, so I think this would be more useful for logging:

```suggestion
				self::NEWSPACK_SCHEDULE_AUTHOR_TERM_BACKFILL,
				! empty( $result->stderr ) ? $result->stderr : $result->stdout,
				[
					'type' => 0 !== $result->return_code || ! empty( $result->stderr ) ? 'error' : 'success',
					'data' => [
						'stdout'    => $result->stdout,
						'timestamp' => gmdate( 'c', time() ),
					],
					'file' => 'newspack_cap_author_terms_backfill',
```

This results in logs like this for success and error, respectively:

```
[2024-10-04 21:41:22] [335]  [newspack_cap_author_term_backfill][Newspack\CLI\Co_Authors_Plus::run_cap_cli_command][SUCCESS]: Found 0 posts with missing author terms.
Processing page 2.
0 records affected
Updating author terms with new counts
Success: Done! - Data: {
    "stdout": "Found 0 posts with missing author terms.\nProcessing page 2.\n0 records affected\nUpdating author terms with new counts\nSuccess: Done!",
    "timestamp": "2024-10-04T21:41:22+00:00"
}

[2024-10-04 21:41:44] [342]  [newspack_cap_author_term_backfill][Newspack\CLI\Co_Authors_Plus::run_cap_cli_command][ERROR]: Error: A fake error occurred! - Data: {
    "stdout": "Found 0 posts with missing author terms.",
    "timestamp": "2024-10-04T21:41:44+00:00"
}
```